packages/client/src/workflow-options.ts

```diff
@@ -37,15 +37,26 @@ export interface WorkflowOptions extends CommonWorkflowOptions {
   followRuns?: boolean;
 }
 
-export interface WorkflowSignalWithStartOptions<SA extends any[] = []> extends WorkflowOptions {
+export type WorkflowSignalWithStartOptions<SignalArgs extends any[] = []> = SignalArgs extends []
```

Comment:
You want the condition to be the other way around.
See https://github.com/temporalio/sdk-typescript/blob/a7be2cd12bf535df607a53035373e33d74c28e25/packages/internal-workflow-common/src/workflow-options.ts#L87

Reply:
Ah, I thought `[]` would just be empty array. I've changed to an array with at least one item:`[any, ...any[]]`